changing DL_Poly units

lexi-x opened the pull request:
Fixes #2393

Changes made in this Pull Request:
<!-- Describe the changes that this PR makes. If applicable, use the following bullet list. --> 
- Added unit conversion to DL_Poly
- ~~Removed forces from TRZ~~
- Updated test values for DL_Poly

As mentioned in #2393, this likely requires more work with test cases and code modifications regarding where the DL_Poly conversions are made, so any feedback is appreciated!

## PR Checklist
<!-- Please use the following checklist to ensure the PR is ready to be reviewed/merged. -->
 - [x] Issue raised/referenced?
 - [x] Tests updated/added?
 - [ ] Documentation updated/added?
 - [ ] `package/CHANGELOG` file updated?
 - [ ] Is your name in `package/AUTHORS`? (If it is not, add it!)

## Developers Certificate of Origin
<!-- 
In order for this PR to be merged you **must certify that you are able to submit your code to be included in MDAnalysis**.
-->

I certify that I can submit this code contribution as described in the [**Developer Certificate of Origin**](https://developercertificate.org/), under the MDAnalysis [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE).


<!-- readthedocs-preview mdanalysis start -->
----
📚 Documentation preview 📚: https://mdanalysis--4983.org.readthedocs.build/en/4983/

<!-- readthedocs-preview mdanalysis end -->

Commit:
modifying force units

Commit:
removing force from TRZ

Commit:
modifying dlpoly test case

Commit:
changing conversion location

Commit:
typo fix

Commit:
fix inconsistencies

Commit:
disregarding TRZ (dropped in 3.0)

lexi-x commented:
Based on feedback about TRZ being dropped, I removed the parts regarding TRZ in the latest commit.

Commit:
reformatting

orbeckst commented:
@cbouy and @talagayev could you please review and shepherd this PR? This is a GSOC applicant for one of your projects.

orbeckst commented on package/MDAnalysis/coordinates/TRZ.py:
```diff
@@ -605,4 +605,4 @@ def close(self):
             return
         self.trzfile.close()
-        self.trzfile = None
+        self.trzfile = None
```
just undo

orbeckst commented on testsuite/MDAnalysisTests/coordinates/test_dlpoly.py:
```diff
@@ -70,8 +70,9 @@ def test_velocities(self, ts):
         ref = np.array([1.056610291, -1.218664448, 3.345828610])
         assert_allclose(ts._velocities[0], ref)
 
+    # modified to original numbers / 100
```
remove comment – we can always look in the git blame to find out when something was changed

orbeckst commented on testsuite/MDAnalysisTests/coordinates/test_dlpoly.py:
```diff
@@ -118,8 +119,9 @@ def test_vel(self, u):
         ref = np.array([2.637614561, 0.5778767520e-01, -1.704765568])
         assert_allclose(u.atoms[2].velocity, ref)
 
+    # modified to original numbers / 100 
```
remove comment

cbouy replied:
@lexi-x removing this would also fix the `linters / black` error here https://github.com/MDAnalysis/mdanalysis/actions/runs/13984105125/job/39154722804?pr=4983#step:5:151

orbeckst commented on package/MDAnalysis/coordinates/DLPoly.py:
```diff
@@ -99,7 +99,7 @@ def _read_first_frame(self):
             velocities = np.array(velocities, dtype=np.float32)
         if has_forces:
-            forces = np.array(forces, dtype=np.float32)
+            forces = np.array(forces, dtype=np.float32) / 100
```
add a comment above this line stating that forces in DL_POLY are 10 J/(mol.Å) and mention issue #2393  — this is weird enough to warrant a hint for future coders

orbeckst replied:
Actually, we should not just divide by 100 here. Instead we should use the normal conversion machinery where you store the native force units in `_DLPOLY_UNITS` and store the conversion factor in units, then use convert_from_native.


See also https://userguide.mdanalysis.org/stable/units.html .

orbeckst reviewed (changes requested):
Good start! See comments. Although dividing by 100 gives the right answer, it is a bit more complicated because we want all our readers to be have in a unified manner. Have a look at other readers such as TRRReader where you define the force units https://github.com/MDAnalysis/mdanalysis/blob/66e7e5c061fa282fe29b6f3da0604be72fd0448c/package/MDAnalysis/coordinates/TRR.py#L150-L151

For DL_POLY, this entry should be `"force": "Da*Angstrom/ps"` in the `_DLPOLY_UNITS` dict.

You then need to add at least one entry for "Da*Angstrom/ps" to `units.forceUnit_factor` https://github.com/MDAnalysis/mdanalysis/blob/66e7e5c061fa282fe29b6f3da0604be72fd0448c/package/MDAnalysis/units.py#L341-L350 with the correct conversion factor. Also update relevant docs and tests.

Then in your code, check if unit conversion is requested (similar to what the TRRReader does in https://github.com/MDAnalysis/mdanalysis/blob/66e7e5c061fa282fe29b6f3da0604be72fd0448c/package/MDAnalysis/coordinates/TRR.py#L213-L214 )


Additionally:
* The history reader should also convert forces
* add a CHANGELOG entry under Fixes
* add a versionchanged 2.10.0 under ConfigReader (and also HistoryReader)

cbouy reviewed (commented):
I second the changes requested by @orbeckst (I wasn't familiar with the unit conversion API to be honest, today I learned!)

lexi-x commented:
@orbeckst @cbouy Thank you for your recommendations! Sorry for the delayed response, I was caught up in making the proposal and another pull request. I made the changes you recommended and I wanted to make sure that my thought process was valid in implementing them before making commits.
1. After changing the DL_POLY force units to Da*Angstrom/ps, I added the conversion factor to units.py based on my understanding of the unit conversions: 
```python
forceUnit_factor = {
    "kJ/(mol*Angstrom)": 1.0,
    "kJ/(mol*A)": 1.0,
    "kJ/(mol*\u212b)": 1.0,
    "kJ/(mol*nm)": 10.0,
    "Newton": 1e13 / constants["N_Avogadro"],
    "N": 1e13 / constants["N_Avogadro"],
    "J/m": 1e13 / constants["N_Avogadro"],
    "kcal/(mol*Angstrom)": 1 / constants["calorie"],
    "Da*Angstrom/ps": 1000.0 / constants["N_Avogadro"] / 1e10 * 1e12
} 
```
2. Regarding making the same unit conversions for History Reader, when I originally added the conversion to both functions, the test cases returned results that were scaled by 10000 instead of 100, so I suspect it was making double conversions. Would this still be an issue when using native force units instead of direct conversion? Thank you!

orbeckst commented:
@lexi-x if all conversions are done in the standard way then there shouldn't be any double conversions. Try it and make the tests pass.

orbeckst commented:
Explain to us why what you're doing is correct — that's the best way to be sure that you're right and makes it easy for us to review.

orbeckst commented:
@lexi-x do you have some time to complete the PR? Would be great to get it done.

orbeckst commented:
@lexi-x do you think you'll continue on the PR?

lexi-x commented:
> @lexi-x do you think you'll continue on the PR?

Yes sorry, I have been a bit busy lately but I'm making the edits right now.

Commit:
issue #2393 fix DLPoly unit conversion

Commit:
merge latest updates

Commit:
merge changes

lexi-x commented:
> @lexi-x do you have some time to complete the PR? Would be great to get it done.

I added the unit conversions and modified the tests accordingly, let me know if anything needs to be changed. Thanks!
